LanguageFeatures/Extension-types/dynamic_semantics_member_invocation_A03_t05.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Consider an invocation of the extension type member m on the
+/// receiver expression e according to the extension type declaration V with
+/// actual type arguments T1, ..., Ts. If the invocation includes an actual
+/// argument part (possibly including some actual type arguments) then call it
+/// args. If the invocation omits args, but includes a list of actual type
+/// arguments then call them typeArgs. Assume that V declares the type variables
+/// X1, ..., Xs
+/// ...
+/// Let Dm be the unique declaration named m that V has.
+///
+/// Evaluation of this invocation proceeds by evaluating e to an object o.
+/// ...
+/// Otherwise, the following is known: args is included, and Dm is a method. The
+/// invocation proceeds to evaluate args to an actual argument list args1. Then
+/// it executes the body of Dm in an environment where this and the name of the
+/// representation are bound in the same way as in the getter invocation, the
+/// type variables of V are bound to the actual values of T1, .. Ts, and the
+/// formal parameters of m are bound to args1 in the same way that they would be
+/// bound for a normal function call. If the body completes returning an object
+/// o2 then the invocation evaluates to o2. If the body throws an object and a
+/// stack trace then the invocation completes throwing the same object and stack
+/// trace.
+///
+/// @description Check invocation of an extension type `call` member
```

Comment:
I can't actually see any extension type members named `call`. There is a function type, `Object Function(num)`, in the bound of `T`, and there are other function types like `Foo`, but I don't see any situations where an invocation of a method named `call` takes place, and said method is a member of an extension type.

That might be hard to achieve anyway, because an extension type can't have `implements T` where `T` is a function type. So it's difficult to find a `call` method anywhere unless we use `id` to look up an actual function.

I suppose it isn't actually worthwhile to try to check the invocation of an extension type `call` method when that `call` method comes from a function type, we can only do it using a callable class or a callable extension type.

Reply:
The description rewritten to reflect that here we are testin method `call` of a representation type. Added `dynamic_semantics_member_invocation_A03_t06.dart` which tests method `call` of an extension type. I think it makes sense to have one test checking that this functionality works for extension types as well

Reply:
OK, thanks!